### PR TITLE
Stm32l152discovery

### PR DESCRIPTION
**Pull request by utzig**
Add initial support for STM32L1xx and STM32L152C-Discovery board.

### CODE REVIEW

**Comment by mkiiskila** [comment]
Is this still WIP?

**Comment by utzig** [comment]
I will remove the comment once is not

**Comment by utzig** [comment]
I2C/SPI are not complete but I plan to merge them with the other STM32 HALoverHALs.

File: kernel/os/src/arch/cortex_m3/os_arch_arm.c [hunk 1]
```diff
@@ -210,7 +210,8 @@ os_arch_os_init(void)
             NVIC->IP[i] = -1;
         }
 
-        NVIC_SetVector(SVCall_IRQn, (uint32_t)SVC_Handler);
+        /* SVC_IRQn or SVCall_IRQn depending on specific model */
+        NVIC_SetVector(-5, (uint32_t)SVC_Handler);
```
Comment (mkiiskila) [inline kernel/os/src/arch/cortex_m3/os_arch_arm.c]: If this indeed varies from MCU to MCU, we could declare a name for this, and set that using a #define inside mcu/mcu.h.

**Pull request merged**